linksharing/handler.go

```diff
@@ -307,3 +336,138 @@ func makeLocation(base *url.URL, reqPath string) string {
 	location.Path = path.Join(location.Path, reqPath)
 	return location.String()
 }
+
+// handleHostingService deals with linksharing via custom URLs.
+func (handler *Handler) handleHostingService(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	host := strings.SplitN(r.Host, ":", 2) // This handles the case where the host contains a port number.
+	serializedAccess, root, err := handler.getRootAndAccess(host[0])
+
+	if err != nil {
+		handler.log.Error("unable to handle request", zap.Error(err))
+		http.Error(w, "unable to handle request", http.StatusInternalServerError)
+		return err
+	}
+
+	access, err := uplink.ParseAccess(serializedAccess)
+	if err != nil {
+		handler.handleUplinkErr(w, "parse access", err)
+		return err
+	}
+
+	project, err := uplink.OpenProject(ctx, access)
+	if err != nil {
+		handler.handleUplinkErr(w, "open project", err)
+		return err
+	}
+	defer func() {
+		if err := project.Close(); err != nil {
+			handler.log.With(zap.Error(err)).Warn("unable to close project")
+		}
+	}()
+
+	// e.g. http://mydomain.com/folder2/index.html with root="bucket1/folder1"
+	rootPath := strings.SplitN(root, "/", 2) // e.g. rootPath=[bucket1, folder1]
+	bucket := rootPath[0]
+	path := strings.TrimPrefix(r.URL.Path, "/") // e.g. path = "folder2/index.html
+	path = strings.TrimSuffix(path, "/")
+	if len(rootPath) == 2 {
+		pathPrefix := strings.TrimPrefix(rootPath[1], "/") // e.g. pathPrefix = "folder1"
+		pathPrefix = strings.TrimSuffix(pathPrefix, "/")
```

Comment:
this also worries me. the object folder//athing is a different object than folder/athing (!). the above construction makes it hard to distinguish.

Reply:
I think that the trimming may not be necessary even, so I'll try to get it to work for the // case

Reply:
How do you create an object `folder//athing`?